### DISCUSSION TITLE
How to make button which suggests user to share his phone number?

### DISCUSSION THREAD

**Discussion by musobekmadrimov** [def-studio/telegraph#483] [Q&A]
Hello, everyone! I am new in this pacakge! 
How to make button which suggests user to share his phone number?
I tried 
`$phone_number = $this->message->contact()->phoneNumber()`
but getting this error:
Call to a member function phoneNumber().

Also I need the correct version of recieving users phone number to save it to DB. 

Please, help!

**Comment by musobekmadrimov** [comment]
Sorry, I found it. But now faced another problem.

```
    public function start(): void
    {
        $name = $this->message->from()->firstName() . " " . $this->message->from()->lastName();
        $this->reply("");
        Telegraph::chat($this->chat)->message("Assalamu alaykum {$name} 👋🏻, sizni ko'rib turganimizdan xursandmiz, iltimos, telefon raqamingizni tasdiqlash maqsadida quyidagi tugmachani bossangiz:")
            ->keyboard(Keyboard::make()->buttons([
                Button::make('✅ Boshlash')->action('getPhoneNumber'),
            ]))->send();
    }

    public function getPhoneNumber(): void
    {
        Telegraph::chat($this->chat)->message("Telefon raqamni ulashish")
            ->replyKeyboard(
                ReplyKeyboard::make()->buttons([
                    ReplyButton::make("🌐 Telefon raqamni ulashish")->requestContact(),
                ])->oneTime(true)->resize()
            )->send();
    }
```
    
From getPhoneNumber method how to assign client phone number? I need do some logic and save it to database! Please, help with this!

**Reply by fabio-ivona** [reply]
After your user clicks on the `🌐 Telefon raqamni ulashish` button, his contact will be sent as message and you can listen to it in your webhook (or through Telegraph getUpdates() method, if now webhook is set)
